### ISSUE TITLE
Moneybird synchronization error: InvalidData("API error 422: {'contact_id': ['cannot be empty']}")

### ISSUE THREAD

**Issue by DeD1rk** [svthalia/concrexit#3294]
https://thalia.sentry.io/issues/4111947935/events/3bc27949720945439e1dc118ffd13aec/?project=1463433

**Comment by DeD1rk** [comment]
This seems to be a consequence of #3293, where creating a 'contact' fails. See https://github.com/svthalia/concrexit/blob/22ff4c23b2373a9baaa0b904bde6ca95ff30f772/website/moneybirdsynchronization/models.py#L192-L198

A `MoneybirdContact` is saved to the database. When pushing it to moneybird fails (that's #3293), no contact_id is saved afterwards, but the MoneybirdContact in the db remains. That's then used creating an external sales invoice which in turn also fails. 

Basically, this issue is caused by incorrect error handling: when something like #3293 occurs, that should not be ignored.

**Comment by DeD1rk** [comment]
This is solved by changes like 
<img width="1169" alt="afbeelding" src="https://github.com/svthalia/concrexit/assets/41264528/6aa90cc9-fa38-4f09-8320-95e1afe2e30d">
 in #3343 and #3338.
